File: single_number.cpp
class Solution {
public:
    int singleNumber(int A[], int n)
    {
        // easy, use the XOR bit operator. Two rules applied here:
        // XOR(A, A) = 0 , XOR(XOR(A, B), A) = B

        int res = A[0], i = 0;
        for (i = 1; i < n; i++)
        {
            res ^= A[i];
        }

        return res;
    }
};

File: Python solutions/single_number.py
class Solution:
    # @param A, a list of integer
    # @return an integer

    # Just use the XOR operator. XOR(A, XOR(B, A)) = B

    def singleNumber(self, A):
        result = A[0]
        for i in A[1:]:
            result ^= i
        return result
